packages/angularjs-auth-passport/lib/webAuthService.js

```diff
@@ -50,26 +51,64 @@ WebAuthService.prototype.getCloudUrl = function() {
 WebAuthService.prototype.getProfile = function() {
   var self = this;
   var url = self.getCloudUrl() + CONSTANTS.PROFILE_URL;
-  return self.http.get(url).then(function(res) {
+  return self.http.get(url)
+  .then(function(res) {
     if (res.data) {
       return res.data;
     }
   }).catch(function(error) {
     self.window.location = self.getCloudUrl() + CONSTANTS.LOGIN_URL;
   });
-}
+};
+
+/**
+ * userProfile setter
+ * @param profile - User profile to be set
+ */
+WebAuthService.prototype.setUserProfile = function(profile) {
+  userProfile = profile;
+};
+
+
+/**
+ * Updates and sets user profile class variable.
+ */
+WebAuthService.prototype.updateUserProfile = function() {
```

Comment:
This is breaking extending Security interface we have which is not desired.
Two options:
- Cache user in getProfile directly
- Move this methods to be outside WebAuthService

Reply:
will do